host/crypto/bcrypt/sha.c

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/host.h>
+#include <openenclave/internal/defs.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/sha.h>
+#include <stdio.h>
+#include <string.h>
+#include "bcrypt.h"
+
+typedef struct _oe_sha256_context_impl
+{
+    BCRYPT_HASH_HANDLE handle;
```

Comment:
Do we need a _magic_ prefix here?
As done for
https://github.com/microsoft/openenclave/blob/bfd18e10c64c3287bbbf25238e594fc9797f4999/enclave/cert.c#L119-L124

Reply:
I don't think so. The use of magic numbers is usually to allow for quick identification of variant types, but the use case introduced seems restricted to distinguishing host vs. enclave implementations* of potentially portable data types (e.g. keys, certs). Context objects (e.g. sha, hmac) in the existing implementation have not traditionally done this, I'm assuming because it's never been kosher to pass these kinds of things between computation boundaries anyway.

I'm not going update the structural use of magic numbers as part of this change set, in particular since @gupta-ak has already called for a broader evaluation of how we are structuring the crypto API in terms of:
* APIs defined for common use between host & enclave
* APIs that need to be implemented inside the enclave to satisfy the OE runtime. 
* APIs that need to be implemented for use in the host library/tools across OS platforms. 

I suspect we'll revisit how we deal with magic numbers as part of that discussion.

\* _Note for example that RSA uses the same magic numbers in both BCrypt & OpenSSL implementations today._

Reply:
Sounds good to me.